ui/hooks/useTokensToSearch.js

```diff
@@ -129,10 +129,14 @@ export function useTokensToSearch({ usersTokens = [], topTokens = {} }) {
       );
       if (
         isSwapsDefaultTokenSymbol(renderableDataToken.symbol, chainId) ||
-        (usersTokensAddressMap[token.address] &&
-          Number(renderableDataToken.balance ?? 0) !== 0)
+        usersTokensAddressMap[token.address]
       ) {
-        tokensToSearchBuckets.owned.push(renderableDataToken);
+        const foundItem = tokensToSearchBuckets.owned.find((item) => {
+          return item.symbol === token.symbol;
+        });
+        if (!foundItem) {
+          tokensToSearchBuckets.owned.push(renderableDataToken);
```

Comment:
This way I won't add a custom token that is already in the `tokensToSearchBuckets.owned` array

Reply:
Nit: It would be nice if there were no duplicates in the array being loop over. Instead of doing this check within the `forEach` loop, we could filter or do a lodash `uniqBy` or something like that on the array before `forEach` is called.

Reply:
Ultimately, I am guessing that we can improve performance (and maybe clean up the code) by using objects instead of arrays to store this data, but that can be a future improvement after this PR.

Reply:
I like the idea to filter it in advance via Lodash, will do that!